### DISCUSSION TITLE
Consistency in signInWithPassword and signInWithOAuth

### DISCUSSION THREAD

**Discussion by bennik88** [supabase/supabase#8873] [Questions]
**Problem:**
When using different sign-up options like emailAndPassword and OAuth there seems to be no consistent way to safe `user_metadata`.

**Options I've looked at**
When saving additional user data like the first_name there is the option to add `options` to the `signUp` function like in the docs to safe `user_metadata`
```js
const { data, error } = await supabase.auth.signUp({
  email: 'example@email.com',
  password: 'example-password',
  options: {
    data: {
      first_name: 'John',
      age: 27,
    },
  },
})
```

But when using different sign-up option like OAuth there seems to be no additional option parameter for data (`user_metadata`) in the official [docs](https://supabase.com/docs/reference/javascript/next/auth-signinwithoauth). Does it mean there is no official way to consistently safe user metadata when multiple sign-up options are present?
I seem to have to skip working with the `user_metadata` and instead have to use/create a related table `profiles` and save all additional data there to be consistent.

If that is the case is there an option to pass additional data to the auth.user signup trigger function that can be caught in a trigger like below to avoid having to make a separate update call from the frontend?
**Frontend**
```js
const { data, error } = await supabase.auth.signInWithOtp({
  email: 'example@email.com',
  // additionalData: { 
  //        first_name: 'John',
  //        age: 27, }
})
```
**Backend trigger**
```sql
begin
 -- Here I would then need to get additional properties that are not present in auth.users like `age`
  insert into public.profiles(id, username)
  values(new.id, new.email);

  return new;
end;
```

**Comment by Benjamin-Dobell** [comment]
Somewhat related, `shouldCreateUser` is also missing from `SignInWithOAuthCredentials`, which is important because there's no separate sign-up method when using OAuth.
